lib/motions/general-motions.coffee

```diff
@@ -107,6 +107,15 @@ class Motion
   isInclusive: ->
     @vimState.mode is 'visual' or @operatesInclusively
 
+  isBeginningOfFile: (cursor) ->
+    cur = cursor.getBufferPosition()
+    not cur.row and not cur.column
+
+  isEndOfFile: (cursor) ->
+    cur = cursor.getBufferPosition()
+    eof = @editor.getEofBufferPosition()
+    cur.row is eof.row and cur.column is eof.column
```

Comment:
Sorry, my bad, it's called isEqual in Point:
https://github.com/atom/text-buffer/blob/master/src/point.coffee#L201
On 5 Aug 2015 20:26, "Matthew Leeds" <notifications@github.com> wrote:

> In lib/motions/general-motions.coffee
> <https://github.com/atom/vim-mode/pull/813#discussion_r36343555>:
>
> > @@ -107,6 +107,15 @@ class Motion
> >    isInclusive: ->
> >      @vimState.mode is 'visual' or @operatesInclusively
> >
> > +  isBeginningOfFile: (cursor) ->
> > +    cur = cursor.getBufferPosition()
> > +    not cur.row and not cur.column
> > +
> > +  isEndOfFile: (cursor) ->
> > +    cur = cursor.getBufferPosition()
> > +    eof = @editor.getEofBufferPosition()
> > +    cur.row is eof.row and cur.column is eof.column
>
> cur.equals(eof) and cur is eof didn't seem to work for me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/vim-mode/pull/813/files#r36343555>.
>